src/Lykke.Tools.AssetMigrator/Implementations/Migrator.cs

```diff
@@ -0,0 +1,169 @@
+            
+            do
+            {
+                IEnumerable<BalanceEntity> balancesBatch;
+                
+                (balancesBatch, continuationToken) = await balanceRepository.GetBalancesAsync
+                (
+                    _options.SourceAssetId,
+                    100,
+                    continuationToken
+                );
+                
+                balances.AddRange(balancesBatch);
+
+            } while (continuationToken != null);
+
+            return balances.Where(x => x.Balance > 0).ToArray();
+        }
+
+        private async Task MigrateAsync(
+            IReadOnlyList<BalanceEntity> balances)
+        {
+            var meClient = new TcpMatchingEngineClient(_options.MEEndPoint, EmptyLogFactory.Instance);
+            var migrationsRepository = new MigrationsRepository(_options.BalancesConncectionString, _options.MigrationId);
+            var operationsRepository = new CashOperationsRepositoryClient(_options.OperationsUrl, EmptyLogFactory.Instance.CreateLog(this), 600);
+            
+            meClient.Start();
+            
+            for (var i = 0; i < balances.Count; i++)
+            {
+                var balance = balances[i];
+                
+                try
+                {
+                    var cashInId = await migrationsRepository.GetOrCreateCashInOutIdAsync(balance.ClientId, _options.TargetAssetId);
+                    var cashOutId = await migrationsRepository.GetOrCreateCashInOutIdAsync(balance.ClientId, _options.SourceAssetId);
+                    var cashinAmount = ((double) balance.Balance * _options.Multiplier).TruncateDecimalPlaces((int) _options.TargetAssetAccuracy);
+                    var cashOutAmount = (double) balance.Balance * -1;
+                    
+                    var cashInResult = await meClient.CashInOutAsync
+                    (
+                        id: cashInId.ToString(),
+                        clientId: balance.ClientId,
+                        assetId: _options.TargetAssetId,
+                        amount: cashinAmount
+                    );
+
+                    if (cashInResult.Status == MeStatusCodes.Ok || cashInResult.Status == MeStatusCodes.Duplicate)
+                    {
+                        var operationId = cashInId.ToString();
+
+                        if (await operationsRepository.GetAsync(balance.ClientId, operationId) == null)
+                        {
+                            await operationsRepository.RegisterAsync(new CashInOutOperation
+                            {
+                                Amount = cashinAmount,
+                                AssetId = _options.TargetAssetId,
+                                ClientId = balance.ClientId,
+                                DateTime = DateTime.UtcNow,
+                                Id = operationId,
+                                TransactionId = operationId,
+                                BlockChainHash = "0x",
```

Comment:
Let's add configuration parameter "HistoryMessage" and pass it as BlockChainHash